New feature - sketch on sheet metal

jaisekjames opened the pull request:
sketch on sheet metal
* Select a face & select a sketch.
It will create cut based on sketch.

![mobile stand 1](https://user-images.githubusercontent.com/44542605/100495484-4646a600-3172-11eb-80df-740d961f33b8.png)
![mobile stand 2](https://user-images.githubusercontent.com/44542605/100495485-4777d300-3172-11eb-90ec-91138466a12b.png)

jaisekjames committed:
Sketch on sheetmetal

sketch on sheetmetal
Select a face 
select a sketch.
It will create cut based on sketch.

jaisekjames committed:
Update InitGui.py

jaisekjames committed:
icon

luzpaz commented:
1. Patched Sheetmetal workbench with new code 
    ```
    cd ~/.FreeCAD/Mod/sheetmetal/
    git fetch 
    curl https://patch-diff.githubusercontent.com/raw/shaise/FreeCAD_SheetMetal/pull/146.patch | git apply
    ```
    Resulted in:
    ```
    patching file SketchOnSheetMetalCmd.py
    patching file InitGui.py
    patching file SketchOnSheet.svg
    ```
2. Fired up FC
3. Switched to sheetmetal wb
Result:
    ```
   05:47:20  name 'SketchOnSheetMetalCmd' is not defined
   05:47:20  Traceback (most recent call last):
    File "<string>", line 61, in Initialize
   ```
```
OS: Manjaro Linux (KDE//usr/share/xsessions/plasma)
Word size of OS: 64-bit
Word size of FreeCAD: 64-bit
Version: 0.19.23074 (Git) AppImage
Build type: Release
Branch: master
Hash: 74b866ed3022fd1452c534dae31a721bcc668f2f
Python version: 3.8.6
Qt version: 5.12.5
Coin version: 4.0.0
OCC version: 7.4.0
Locale: English/United States (en_US)
```

luzpaz committed:
Added missing import

Still doesn't fix the issue in https://github.com/shaise/FreeCAD_SheetMetal/pull/146#issuecomment-735216604

luzpaz commented on InitGui.py:
```diff
@@ -58,8 +58,9 @@ def Initialize(self):
         import SheetMetalRelief
         import SheetMetalJunction
         import SheetMetalBend
+        SketchOnSheetMetalCmd
```
doesn't this need to be `import SketchOnSheetMetalCmd`  
if so then I've submitted a PR https://github.com/jaisekjames/FreeCAD_SheetMetal/pull/1

jaisekjames replied:
Thank you.

luzpaz replied:
np, but it still doesn't fix the issue :disappointed:

jaisekjames replied:
I just downloaded my "new feature" branch zip & pasted on mod folder. It is working. No error.

luzpaz replied:
my bad. I wasn't applying the patch correctly. I needed to use `git apply` (i've amended the steps above (https://github.com/shaise/FreeCAD_SheetMetal/pull/146#issuecomment-735216604) to reflect that

jaisekjames committed:
Merge pull request #1 from luzpaz/new_feature_fix

Added missing import

luzpaz commented:
@jaisekjames BTW, when I added the patch I get this output:
```
<stdin>:25: trailing whitespace.
#  
<stdin>:27: trailing whitespace.
#  
<stdin>:32: trailing whitespace.
#  
<stdin>:37: trailing whitespace.
#  
<stdin>:42: trailing whitespace.
#  
warning: squelched 36 whitespace errors
warning: 41 lines add whitespace errors.
```
Do you mind cleaning up the trailing whitespace?

jaisekjames committed:
white spaces

shaise commented:
Thanks for this great feature!
